src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -683,6 +683,7 @@ private static void CheckInlineArrayTypeIsSupported(SyntaxNode syntax, TypeSymbo
                 }
 
                 var elementConversions = conversion.UnderlyingConversions;
+                conversion.MarkUnderlyingConversionsChecked();
```

Comment:
>conversion.MarkUnderlyingConversionsChecked(); [](http://example.com/codeflow?start=16&length=46)

Consider moving this after the work of checking the underlying conversions is done. I assume that is after the loop below.

Reply:
BTW, it is not clear to me what is this operation going to change. It look like we are mutating a local value of a struct, which is not returned or stored anywhere.

Reply:
> it is not clear to me what is this operation going to change

I guess this operation is mutating a class instance referenced by the struct, therefore, the mutation is preserved even when the struct value is not preserved.
Is this change backed by a test?